navpat.py

```diff
@@ -195,27 +205,24 @@ def navigate_pattern(metalog, playground, viewer=None):
         can = CAN()
         can.relog(can_log_name, timestamps_log=open(metalog.getLog('timestamps'), 'w'))
     can.resetModules(configFn=setup_faster_update)
-    robot = JohnDeere(can=can, localization=SimpleOdometry())
+
+    if conf is not None and 'localization' in conf.data:
+        loc = SimpleOdometry.from_dict(conf.data['localization'])
+    else:
+        loc = SimpleOdometry(pose = (0.0, 2.5, 0.0))
+        loc.global_map = [(0.0, 0.0), (15.0, 0.0), (15.0, 5.0), (0.0, 5.0)]
+
+    jd_config = None
+    if conf is not None and 'johndeere' in conf.data:
+        jd_config = conf.data['johndeere']
+    robot = JohnDeere(can=can, localization=loc, config=jd_config)
```

Comment:
A-ha. This is the part I was confused about and why I kept saying navpat "interprets" the localization key. This is what I would do:

```python
loc = SimpleOdometry(config=conf.data.get('localization'))
robot = JohnDeere(can=can, localization=loc, config=conf.data.get('johndeere'))
```

In a perfect world one could just do
```python
module = importlib.import_module(name)
klass = getattr(module, name)
instance = klass(conf.get(name))
```
Or something like that. 

But we can leave that for the future "launcher refactoring" thing we have been talking about.